### PR TITLE
fix(reflux): fix types so we stop casting

### PR DESCRIPTION
**Pull request by JonasBa**
Less type casting, less chance for errors

### CODE REVIEW

File: static/app/types/reflux.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {SafeRefluxStore, SafeStoreDefinition} from 'sentry/utils/makeSafeRefluxStore';
+import {Store, StoreDefinition} from 'reflux';
+
+declare module 'reflux' {
+  function createStore<T extends SafeStoreDefinition | StoreDefinition>(
```
Comment (evanpurkhiser) [inline static/app/types/reflux.d.ts:5]: I'm happy w/ this.

It does expose some internals right? I don't think I care though

**Review by evanpurkhiser — APPROVED** [review]
nice work!

File: static/app/types/reflux.d.ts (hunk 1 above)
Comment (JonasBa) [reply static/app/types/reflux.d.ts:5]: Yeah, same thing. If you check the final type it's pretty much the same + it'll add the safe store definition if you are passing through that helper
CleanShot 2022-04-01 at 15 35 21@2x

Comment (evanpurkhiser) [reply static/app/types/reflux.d.ts:5]: You'll end up being able to retrieve some internal values though now right. Like if it had a `Internal` `state` you can do `.state` now on the store 🤷‍♂️

Comment (JonasBa) [reply static/app/types/reflux.d.ts:5]: Yeah, for sure. CommiterStoreDefinition extends Internal and CommonStoreDefinition

**Pull request merged**